atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -205,8 +204,18 @@ boolean validateLocksOnReads() {
         return Callback.noOp();
     }
 
-    public static ImmutableTransactionManagers.ConfigBuildStage builder() {
-        return ImmutableTransactionManagers.builder();
+    public interface Configurator<T extends TransactionManager> {
+        Supplier<T> configure(Builder builder);
+    }
+
+    public static class Builder extends ImmutableTransactionManagers.Builder {
+        <T extends TransactionManager> Supplier<T> serializable(Configurator<T> configurator) {
+            return configurator.configure(this);
```

Comment:
We could prevent an off scenario by invalidating the builder after calling configurator:

```
public static class Builder extends ImmutableTransactionManagers.Builder {
        private boolean valid = true;
        <T extends TransactionManager> Supplier<T> serializable(Configurator<T> configurator) {
            T value = configurator.configure(this);
            this.valid = false
            return value;
         }

        @Override
        public ImmutableTransactionManagers build() {
          if (!valid) { throw new InvalidStateException blah }
          return super.build();
        }
```

Reply:
But maybe not important since we already allow the footgun of:

```
TM.Builder builder = ...
tm1 = builder.build().serializable();
tm2 = builder.build().serializable();
```